ros2launch/ros2launch/api/api.py

```diff
@@ -154,11 +186,75 @@ def launch_a_launch_file(*, launch_file_path, launch_file_arguments, debug=False
             launch_arguments=parsed_launch_arguments,
         ),
     ])
+    if secure is not None:
+        # declare keystore so if it is a TemporaryDirectory, it gets cleaned up when out of scope
```

Comment:
Doesn't it technically go out of scope as soon as it leaves this `if` block?

Reply:
Nope, blocks don't limit scope in python

Reply:
Ah indeed. Very good then.